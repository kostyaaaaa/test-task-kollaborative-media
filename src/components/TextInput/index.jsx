import { func, oneOf, string } from 'prop-types';
import React from 'react';

const TextInput = ({ type, id, className, placeholder, onChange, value }) => {
  return (
    <input
      type={type}
      id={id}
      className={className}
      placeholder={placeholder}
      onChange={onChange}
      value={value}
    />
  );
};

TextInput.defaultProps = {
  id: '',
  className: '',
  value: '',
};

TextInput.propTypes = {
  type: oneOf(['text', 'email', 'tel']).isRequired,
  id: string,
  className: string,
  placeholder: string.isRequired,
  onChange: func.isRequired,
  value: string,
};

export default TextInput;
