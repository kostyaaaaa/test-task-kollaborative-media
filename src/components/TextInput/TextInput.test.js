import renderer, { act } from 'react-test-renderer';
import TextInput from './index';

const mockedOnChange = jest.fn();

const component = renderer.create(
  <TextInput
    type="text"
    id="mocked-id"
    placeholder="mocked-placeholder"
    onChange={mockedOnChange}
    className="text-xl"
  />,
);

it('should match snapshot', () => {
  expect(component).toMatchSnapshot();
});

it('should apply provided props', () => {
  expect(component.toTree().rendered.props.className).toEqual('text-xl');
  expect(component.toTree().rendered.props.type).toEqual('text');
  expect(component.toTree().rendered.props.id).toEqual('mocked-id');
  expect(component.toTree().rendered.props.placeholder).toEqual(
    'mocked-placeholder',
  );
});

it('should call provided onChange prop', () => {
  expect(mockedOnChange).toHaveBeenCalledTimes(0);
  // Simulate a change event
  const mEvent = { target: { value: 'mocked value' } };
  act(() => {
    component.root.props.onChange(mEvent);
  });
  expect(mockedOnChange).toHaveBeenCalledTimes(1);
});
