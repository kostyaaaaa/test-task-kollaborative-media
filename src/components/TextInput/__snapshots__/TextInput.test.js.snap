// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<input
  className="text-xl"
  id="mocked-id"
  onChange={[MockFunction]}
  placeholder="mocked-placeholder"
  type="text"
  value=""
/>
`;
